src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -264,19 +259,20 @@ def readClusterSettings(self) -> None:
         # Determine where to deploy workers.
         self._worker_subnets_by_zone = self._get_good_subnets_like(self._leader_subnet)
 
-        from boto.utils import get_instance_metadata
-
-        self._keyName = instance["KeyName"]
-        logger.error(f"{self._keyName}")
-        # self._keyName = list(instanceMetaData['public-keys'].keys())[0]
         self._leaderPrivateIP = ec2_metadata.private_ipv4  # this is PRIVATE IP
         self._tags = {k: v for k, v in (self.getLeader().tags or {}).items() if k != _TAG_KEY_TOIL_NODE_TYPE}
         # Grab the ARN name of the instance profile (a str) to apply to workers
-        leader_info = ec2_metadata.iam_info
+        leader_info = None
+        for attempt in old_retry(timeout=300, predicate=lambda e: isinstance(e, ValueError)):
+            with attempt:
+                leader_info = ec2_metadata.iam_info
+                if leader_info is None:
+                    raise ValueError(f"Failed to query leader IAM metadata. EC2 instance is not reachable.")
```

Comment:
This seems a little like a misuse of `ValueError`. Will `ec2_metadata` raise ValueError when it fails?

If you're using `old_retry` I think you can just go around the loop again and generate a new `attempt` if you don't get an exception but also don't like the answer.

Also, don't you need a `break` to get out of this loop when you do have the metadata?

Also also, I don't think "EC2 instance is not reachable" is the right message here. At this point *we're* the EC2 instance. Were trying to talk to the IMDS to get metadata about ourselves. It's odd to say that the machine we're on is not reachable by the service we assume is supposed to be talking to us.